HW1/P4/P4.py

```diff
@@ -0,0 +1,47 @@
+import findspark
+findspark.init()
+import pyspark
+import csv
+
+from P4_bfs import bfs
+
+
+def construct_edges(x):
+    # add each edge to list  
+    i, s_list = x
+    l = len(s_list)
+    edges = []
+    for i in range(l):
+        for j in range(l):
+            if i != j: 
+                edges.append(((s_list[i], s_list[j]), None))
+    return edges
+
+def main():
+    # queries are CAPTAIN AMERICA 
+    # MISS THING/MARY
+    # ORWELL
+    sc = pyspark.SparkContext()
+    sc.setLogLevel('WARN')
+    source_reader = csv.reader(open("source.csv", 'rb'), delimiter=',')
```

Comment:
Why did you not use the sc.textFile instead of csv.reader here? sc.textFile parallelize the data for you, so it'll be faster than having the csv.reader